ffb-server/src/main/java/com/fumbbl/ffb/server/RosterCache.java

```diff
@@ -28,7 +44,17 @@ public RosterCache() {
 		rosterFileById = new HashMap<>();
 	}
 
-	public Roster getRosterById(String rosterId, Game game) {
+	public Roster getRosterForTeam(Team team, Game game) {
+		// In newer versions of the XML format, the `<rosterId>` is not used (but is still present).
+		// So we first check for the presence of a roster matching the team id, and only if no roster
+		// is found, do we fall back to looking up the roster using the original rosterId.
+		String rosterId = team.getId();
+		if (!rosterFileById.containsKey(rosterId)) {
```

Comment:
`rosterFileById` is still populated in the same way as before, isn't it? So it only contains the value of the id attribute of the roster tag as keys. 

How can `contains` call for a team id ever return true in this case?

Reply:
Good catch. I was accidentially testing on a roster file I modified, rather than the direct API output 🙈. I will rework this.

Reply:
I fixed the problem and pushed a fix. I'm unsure how you normally merge branches, so I force-pushed the change to keep the history clean.